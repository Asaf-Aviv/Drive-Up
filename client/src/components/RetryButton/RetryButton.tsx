import React from 'react';
import { Button } from '@material-ui/core';

interface RetryButtonProps {
  onClick(): void;
}

const RetryButton: React.FC<RetryButtonProps> = ({ onClick }) => (
  <Button onClick={onClick}>
    Retry
  </Button>
);

export default RetryButton;
