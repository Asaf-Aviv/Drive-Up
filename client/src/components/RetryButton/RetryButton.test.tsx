import React from 'react';
import { shallow, mount } from 'enzyme';
import RetryButton from './RetryButton';

const props = { onClick: jest.fn() };

describe('RetryButton component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<RetryButton onClick={props.onClick} />);

    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render a button', () => {
    const wrapper = mount(<RetryButton onClick={props.onClick} />);
    const button = wrapper.find('button');

    expect(button.length).toEqual(1);
  });

  it('should fire onClick prop function on click', () => {
    const wrapper = mount(<RetryButton onClick={props.onClick} />);
    wrapper.find('button').simulate('click');

    expect(props.onClick).toBeCalledTimes(1);
  });
});
