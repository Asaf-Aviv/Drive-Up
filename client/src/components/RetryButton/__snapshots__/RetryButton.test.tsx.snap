// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetryButton component renders correctly 1`] = `
"<WithStyles(ForwardRef(Button)) onClick={[Function: mockConstructor]}>
  Retry
</WithStyles(ForwardRef(Button))>"
`;
